starknet_py/net/full_node_client.py

```diff
@@ -474,11 +473,32 @@ def _create_broadcasted_txn_common_properties(transaction: AccountTransaction) -
         "type": "INVOKE"
         if transaction.tx_type == TransactionType.INVOKE_FUNCTION
         else transaction.tx_type.name,
-        "max_fee": hash_to_felt(transaction.max_fee),
-        "version": hash_to_felt(transaction.version),
-        "signature": [hash_to_felt(sig) for sig in transaction.signature],
-        "nonce": hash_to_felt(transaction.nonce)
+        "max_fee": _to_rpc_felt(transaction.max_fee),
+        "version": _to_rpc_felt(transaction.version),
+        "signature": [_to_rpc_felt(sig) for sig in transaction.signature],
+        "nonce": _to_rpc_felt(transaction.nonce)
         if transaction.nonce is not None
-        else None,
+        else "0x00",
```

Comment:
Specification requires the nonce to always be there (which doesn't make sens in tx version 0). I had to introduce a default value in place of `None`. 

This is not ideal, but tx version 0 will be gone soon anyways.